### PR TITLE
ConnectionListener test for JBJCA-704

### PR DESCRIPTION
**Pull request by hsvabek**
Test for https://issues.jboss.org/browse/JBJCA-704

### CODE REVIEW

**Comment by jesperpedersen** [comment]
Missing Statement.close() + ResultSet.close()

**Comment by hsvabek** [comment]
retest this please

File: testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/jca/connectionlistener/AbstractTestsuite.java [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+    	try {
+    		if(moduleJar != null){
+    			moduleJar.delete();
+    		}
+		} catch (Exception e) {
+			log.warn(e.getMessage());
+		}
+    }
+
+    public void addModule(final String moduleName) throws Exception {
```
Comment (ctomc) [inline testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/jca/connectionlistener/AbstractTestsuite.java]: why so much duplication of test code? this is the 4th (5th) time jca tests add the same code, and we need to fix it on 5 places if(when) it gets broken.
Take a look at AbstractModuleDeploymentTestCaseSetup which does pretty much all this already. and is also only used for JCA tests

Comment (hsvabek) [inline testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/jca/connectionlistener/AbstractTestsuite.java]: I know about this class. But I did not want to add dependency on module (wildfly-ts-integ-basic) with this class to another module (wildfly-ts-integ-manualmode).

What solution do you suggest?

Comment (ctomc) [inline testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/jca/connectionlistener/AbstractTestsuite.java]: I'm not completely sure this needs to be in manual mode testsuite. If it does, the AbstractModuleDeploymentTestCaseSetup (at least most of it) could  moved to testsuite-shared

as code for deploying/creating modules is quite common and even jca is not the only one using it.

**Comment by hsvabek** [comment]
retest this please

**Comment by hsvabek** [comment]
retest this please

**Comment by hsvabek** [comment]
retest this please

**Comment by hsvabek** [comment]
Missing Statement.close() + ResultSet.close() was added
I moved some code from AbstractModuleDeploymentTestCaseSetup to testsuite-shared

**Comment by stuartwdouglas** [comment]
This commit should reference a WFLY JIRA not an EAP7 JIRA.

**Comment by hsvabek** [comment]
retest this please

File: testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/jca/connectionlistener/AbstractTestsuite.java [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+            return address;
+        }
+    }
+    
+    protected void modifyModuleIronJacamarJdbc() throws Exception{
+    	addModule(moduleDefaultPath);
+    	createSubModuleJarConnectionListenerImpl(JAR_NAME);
+    	
+    	//we have to edit module.xml and add our new JAR as resource
```
Comment (bstansberry) [inline testsuite/integration/manualmode/src/test/java/org/jboss/as/test/manualmode/jca/connectionlistener/AbstractTestsuite.java]: If this is the case, IMHO the feature is broken.

Plugin points exposed via classname attributes in the WildFly management model should also include an attribute that allows the user to specify the name of the module in which the class resides, and the subsystem should load the class from that module.

If a user alters one of our modules, a patch affecting that module will not apply.

**Comment by hsvabek** [comment]
I will create new PR for this.

**Pull request closed without merging**